lib/mix/lib/mix/tasks/release.init.ex

```diff
@@ -87,8 +87,10 @@ defmodule Mix.Tasks.Release.Init do
     #!/bin/sh
     set -e
 
-    SELF=$(readlink "$0" || true)
+    SELF=$(readlink -f "$0" || true)
     if [ -z "$SELF" ]; then SELF="$0"; fi
+    if [ "${SELF#/}" = "$SELF" ]; then SELF="$(cd "$(dirname "$0")" && pwd -P)/$(basename "$SELF")"; fi
+    SELF=$(cd "$(dirname "$SELF")" && pwd -P)/$(basename "$SELF")
```

Comment:
If we want to address this, then we need to make it a recursive function. You can check out how it is done in `bin/elixir` in this project.

Reply:
You’re right! I was using `readlink -f` to recursively follow the symlinks, but I realized it doesn’t work on macOS and other BSD-based systems.

I’ve now ported the function used in `bin/elixir` to handle this correctly. Additionally, I adjusted the function to return to the original directory afterward, similar to how the .bat version of the script handles it with `pushd` and `popd`. Just to confirm, is this approach okay?